code/modules/mob/living/living_defense.dm

```diff
@@ -125,6 +125,8 @@
 		hit_impact(P.get_structure_damage(), hit_dir)
 		for(var/damage_type in P.damage_types)
 			var/damage = P.damage_types[damage_type]
+			if(faction in P.supereffective_factions)
+				damage *= P.supereffective_mult
```

Comment:
Hey, is there any way you could get this supereffective mult outside of our bullet_act proc? An O(n) search loop here will probably slow things down a little.

Reply:
I can definitely do something like
```
if(LAZYLEN(P.supereffective_factions))
    loop
```